### ISSUE TITLE
Fine tune BioBERT on multiple new categories for NER task

### ISSUE THREAD

**Issue by kormilitzin** [dmis-lab/biobert#14]
Thank you for this amazing implementation. I'm wondering, how tricky it is possible to extend (by fine tuning on specific corpora) IOB tags to new multiple categories?

For example, I would to fine tune BioBERT on my corpus with three entity categories (a, b, c). Normally, I would label my data as [B-a, I-a, B-b, I-b, B-c, I-c, O, X, [CLS], [SEP]], however, when I tried to modify your run_ner.py, it fails.

Any suggestions how to implement multiple NER taks with BioBERT?

**Comment by gccome** [comment]
Hi @kormilitzin . I am also using this library and answer to your question is yes. Please see issue #12 . you need to modify some numbers  there.

**Comment by kormilitzin** [comment]
@gccome, many thanks for your reference. However, now I'm facing a different problem: the incompatibility of the pretrained graph (with size 7: number of tokens +1) and a new graph with size 11. 


> InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
> 
> Assign requires shapes of both tensors to match. lhs shape= [11,768] rhs shape= [7,768]


How did you manage to overcome the problem?

**Comment by gccome** [comment]
Hi @kormilitzin hmm that's strange. I didn't encounter the issue you have here. Theoretically, the pretrained BioBERT is a language model which should not contain the number of labels anywhere in the graph, you can fine tune NER task with any number of entities. I can successfully ran finetuning on my own NER data, but my issue is at entity level evaluation.

**Comment by wonjininfo** [comment]
Hi @kormilitzin, 
As @gccome mentioned, our pre-trained BioBERT (pre-trained weights and graph) is a language model so it doesn't contains the number of labels. (Thanks @gccome !!)

However, if you want to evaluate multi-label NER results, you need to modify  `ner_detokenizer.py`.
Released version of `ner_detokenizer.py` only supports detokenization of single-label tasks. 
https://github.com/dmis-lab/biobert/issues/12#issuecomment-468494350

Thank you

**Comment by MeRajat** [comment]
Maybe this can help.  https://github.com/MeRajat/SolvingAlmostAnythingWithBert . It's pytorch implementation.

**Comment by rahul-1996** [comment]
> Maybe this can help. https://github.com/MeRajat/SolvingAlmostAnythingWithBert . It's pytorch implementation.
Thank you so much, this conversion saved us a lot of work. Do let us know if there is any paper etc that you're writing/ have written that we can cite, thanks again!

**Comment by MeRajat** [comment]
> > Maybe this can help. https://github.com/MeRajat/SolvingAlmostAnythingWithBert . It's pytorch implementation.
> > Thank you so much, this conversion saved us a lot of work. Do let us know if there is any paper etc that you're writing/ have written that we can cite, thanks again!

Not yet started writing anything. Maybe in future :)
